### PR TITLE
Implement new JS client

### PR DESCRIPTION
**Pull request by stravid**

### CODE REVIEW

File: vendor/assets/javascripts/mountable_file_server.js [hunk 1]
```diff
@@ -1,61 +1,44 @@
...
+  var MountableFileServerUploader = function(options) {
+    return {
+      url: options.url,
+      uploadFile: function(options) {
+
+        var file = options.file;
+        var type = options.type;
+        var onStart = options.onStart || function() {};
+        var onProgress = options.onProgress || function() {};
+        var onSuccess = options.onSuccess || function() {};
```
Comment (tombu) [inline vendor/assets/javascripts/mountable_file_server.js]: Just another thought out of the blue: Is there a use case for a callback `onError`? If yes, we could add it.

**Comment by tombu** [comment]
Good job once again. I´m looking forward to using this new API :)

While reviewing I thought about uploading with the ruby client, e.g. after when generating a document on the server – do you already have that on your radar? Just another spontaneous thought. Sorry for those confusions all the time :-/

File: vendor/assets/javascripts/mountable_file_server.js (hunk 1 above)
Comment (stravid) [inline vendor/assets/javascripts/mountable_file_server.js]: I guess there is a use case, it's just not in this scope. Let's create a Trello ticket.

**Comment by stravid** [comment]
Same as the other issue, uploading files via the Ruby client is a new ticket.

**Pull request merged**